### PR TITLE
Add io.github.srb2.rphys

### PR DESCRIPTION
**Pull request by katsy**
This is my custom mod of Sonic Robo Blast 2, which already exists on flathub as an app. I'v never really used flatpak before but people have been asking for one so they can play my mod easier so I figured I'd give it a shot. I've tested compile and it runs fine with `flatpak run` but let me know if anything needs to be changed.

### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://docs.flathub.org/docs/for-app-authors/requirements
[maint]: https://docs.flathub.org/docs/for-app-authors/maintanance
[submission]: https://docs.flathub.org/docs/for-app-authors/submission

### CODE REVIEW

File: org.rphys.srb2.yml [hunk 1]
```diff
@@ -0,0 +1,101 @@
+app-id: org.rphys.srb2
```
Comment (bbhtt) [inline org.rphys.srb2.yml:1]: `rphys.org` doesn't seem resolve anywhere. See https://docs.flathub.org/docs/for-app-authors/requirements#application-id

> The Application ID should be a real URL of a domain that the app author has control over or where their app is hosted.

File: rphys.sh [hunk 2]
```diff
@@ -0,0 +1 @@
+#!/bin/sh
```
Comment (bbhtt) [inline rphys.sh:1]: Seems empty?

Comment (katsy) [reply rphys.sh:1]: The tutorial said to include it but I can delete if it's not needed.

File: com.github.rphys.yml [hunk 3]
```diff
@@ -0,0 +1,101 @@
+app-id: com.github.rphys
```
Comment (bbhtt) [inline com.github.rphys.yml:1]: Cannot be `com.github`, you don't control that domain. See https://docs.flatpak.org/en/latest/conventions.html#application-ids

File: org.rphys.srb2.yml (hunk 1 above)
Comment (katsy) [reply org.rphys.srb2.yml]: I changed the app name to use github.com. Should I open a new PR with the branch name fixed or keep using this one?

File: com.github.rphys.yml (hunk 3 above)
Comment (katsy) [reply com.github.rphys.yml:1]: "The Application ID should be a real URL of a domain that the app author has control over or where their app is hosted."

I don't control github, but it's where my app is hosted. I do not control any domains. Can I just not submit an app?

File: com.github.rphys.desktop [hunk 4]
```diff
@@ -0,0 +1,11 @@
...
+Version=1.0
+Type=Application
+Name=srb2 rphys
+Comment=Gameplay overhaul mod for Sonic Robo Blast 2
+TryExec=rphys
+Exec=rphys
+Icon=com.github.rphys
+Terminal=false
+Categories=Game;
+Name[en_US]=com.github.rphys.desktop
```
Comment (bbhtt) [inline com.github.rphys.desktop:11]: This is wrong.

File: com.github.rphys.yml [hunk 5]
```diff
@@ -0,0 +1,101 @@
...
+      - type: archive
+        url: https://lib.openmpt.org/files/libopenmpt/src/libopenmpt-0.7.3+release.autotools.tar.gz
+        sha256: 2cf8369b7916b09264f3f14b9fb6cef35a6e9bee0328dec4f49d98211ccfd722
+        x-checker-data:
+          type: anitya
+          project-id: 141468
+          stable-only: true
+          url-template: https://lib.openmpt.org/files/libopenmpt/src/libopenmpt-$version+release.autotools.tar.gz
+
+  - name: srb2
```
Comment (bbhtt) [inline com.github.rphys.yml:54]: An appdata is needed.

File: com.github.rphys.yml (hunk 3 above)
Comment (bbhtt) [reply com.github.rphys.yml:1]: > While a project may be hosted on GitHub or GitLab it does not have any control over the github.com or gitlab.com domain. Instead you should use io.github or io.gitlab as shown above.

Use `io.github.<app>`

File: rphys.sh (hunk 2 above)
Comment (katsy) [reply rphys.sh]: deleted it

File: com.github.rphys.yml (hunk 3 above)
Comment (katsy) [reply com.github.rphys.yml]: done

File: com.github.rphys.desktop (hunk 4 above)
Comment (katsy) [reply com.github.rphys.desktop]: Assuming you meant the .desktop, I fixed that

File: com.github.rphys.yml (hunk 5 above)
Comment (katsy) [reply com.github.rphys.yml]: Added

File: com.github.rphys.desktop (hunk 4 above)
Comment (bbhtt) [reply com.github.rphys.desktop]: No, why is the name, the app-id? It should be the specific name of the application like "srb2 rphys"

Comment (katsy) [reply com.github.rphys.desktop]: Ohm I didn't intentionally add that second Name line, it just appeared in the file at some point. I'll delete it,

File: io.github.srb2.rphys.desktop
Comment (bbhtt) [inline io.github.srb2.rphys.desktop:1]: This ideally should belong in your project repo.

File: io.github.srb2.rphys.yml [hunk 6]
```diff
@@ -0,0 +1,103 @@
...
+          - export SRB2WADDIR=/app/extra
+          - lsdl2srb2 "$@"
+        dest-filename: rphys.sh
+
+  - name: srb2-assets
+    buildsystem: simple
+    build-commands:
+      - install -t $FLATPAK_DEST/bin apply_extra
+    sources:
+      - type: extra-data
```
Comment (bbhtt) [inline io.github.srb2.rphys.yml:87]: Why is this extra-data? Both, seems to be open source

File: io.github.srb2.rphys.yml [hunk 7]
```diff
@@ -0,0 +1,103 @@
...
+            ARCH_MAKE_ARGS: LINUX64=1
+        x86_64:
+          env:
+            ARCH_MAKE_ARGS: LINUX64=1 X86_64=1
+    build-commands:
+      - make -C src -j $FLATPAK_BUILDER_N_JOBS NOUPX=1 SDL=1 $ARCH_MAKE_ARGS
+      - install -D -m 755 -t $FLATPAK_DEST/bin bin/lsdl2srb2
+      - install -D -m 644 srb2.png $FLATPAK_DEST/share/icons/hicolor/256x256/apps/$FLATPAK_ID.png
+      - install -D -m 755 rphys.sh $FLATPAK_DEST/bin/srb2rphys
+      - install -D -m 644 -t $FLATPAK_DEST/share/applications $FLATPAK_ID.desktop
```
Comment (bbhtt) [inline io.github.srb2.rphys.yml:69]: You forgot to add the desktop and metainfo as file sources.

File: io.github.srb2.rphys.yml [hunk 8]
```diff
@@ -0,0 +1,103 @@
+app-id: com.github.rphys
```
Comment (bbhtt) [inline io.github.srb2.rphys.yml:1]: Change app id here too

File: io.github.srb2.rphys.yml [hunk 9]
```diff
@@ -0,0 +1,103 @@
...
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+command: srb2
+finish-args:
+  - --share=ipc
+  - --share=network
+  - --socket=fallback-x11
+  - --socket=pulseaudio
+  - --socket=wayland
```
Comment (bbhtt) [inline io.github.srb2.rphys.yml:11]: The game in GNOME wayland looks like this:

- No window decoration to drag
- No fullscreen

<details>
<summary>image</summary>

image
</details>

Is wayland actually supported? If not just use `--socket=x11`

File: io.github.srb2.rphys.yml [hunk 10]
```diff
@@ -0,0 +1,103 @@
...
+  - --device=all
```
Comment (bbhtt) [inline io.github.srb2.rphys.yml:12]: Why? Seems like it could do with just `dri`?

File: io.github.srb2.rphys.desktop
Comment (hfiguiere) [reply io.github.srb2.rphys.desktop:1]: There is already one https://github.com/katsy/srb2/blob/rphys_v7/debian-template/srb2.desktop

And it should be used (and also fixed upstream)

File: io.github.srb2.rphys.yml (hunk 9 above)
Comment (katsy) [reply io.github.srb2.rphys.yml:11]: Wayland is supported. I'm not sure why the game looks like that on your system but I haven't touched SDL2 or rendering in my mod. The same issue should happen on the vanilla game's flatpak https://flathub.org/apps/org.srb2.SRB2

File: io.github.srb2.rphys.yml (hunk 6 above)
Comment (katsy) [reply io.github.srb2.rphys.yml:87]: The asset files are not open source and not bundled with the source code, this was also discussed in https://github.com/flathub/flathub/pull/1489

File: io.github.srb2.rphys.yml (hunk 7 above)
Comment (katsy) [reply io.github.srb2.rphys.yml:69]: fixed

File: io.github.srb2.rphys.yml (hunk 8 above)
Comment (katsy) [reply io.github.srb2.rphys.yml]: done

File: io.github.srb2.rphys.yml [hunk 11]
```diff
@@ -0,0 +1,103 @@
+app-id: com.github.rphys
+runtime: org.freedesktop.Platform
+runtime-version: '23.08'
+sdk: org.freedesktop.Sdk
+command: srb2
```
[the remark this answers is not in the document]
Comment (katsy) [reply io.github.srb2.rphys.yml]: i changed it

File: io.github.srb2.rphys.desktop
Comment (katsy) [reply io.github.srb2.rphys.desktop:1]: The debian template is not maintained upstream and has gone entirely unused for a long while. I just deleted that directory and added the files to the project.

**Comment by hfiguiere** [comment]
bot, build io.github.srb2.rphys

**Comment by bbhtt** [comment]
> It starts with this ridiculously small window size for me and I cannot resize it manually, regardless of video resolution settings:

I mentioned it https://github.com/flathub/flathub/pull/4547#discussion_r1336753735 the answer was that it supports wayland.

**Comment by katsy** [comment]
Changed to use x11 only until upstream can fix the problem and filed an issue on the vanilla game's Flathub repo that they need to remove the Wayland socket https://github.com/flathub/org.srb2.SRB2/issues/59

**Comment by pizzadude** [comment]
The manifest was missing the "  - --persist=.srb2" line, therefore saving didn't work. This pull request fixes it:

https://github.com/flathub/io.github.srb2.rphys/pull/2

**Pull request closed without merging**